ovhcli/domain/list.go

```diff
@@ -25,15 +27,32 @@ var cmdDomainList = &cobra.Command{
 		common.Check(err)
 
 		if withDetails {
-			domainsComplete := []ovh.Domain{}
-			for _, domain := range domains {
-				d, err := client.DomainInfo(domain.Domain)
-				common.Check(err)
-				domainsComplete = append(domainsComplete, *d)
-			}
-			domains = domainsComplete
+			domains = getDetailledDomainList(client, domains)
 		}
 
 		common.FormatOutputDef(domains)
 	},
 }
+
+func getDetailledDomainList(client *ovh.Client, domains []ovh.Domain) []ovh.Domain {
+	var wg sync.WaitGroup
+	wg.Add(len(domains))
+
+	domainsChan := make(chan ovh.Domain)
+	domainsComplete := []ovh.Domain{}
+	for _, domain := range domains {
+		go func(domain ovh.Domain) {
+			d, err := client.DomainInfo(domain.Domain)
+			common.Check(err)
+			domainsChan <- *d
+		}(domain)
+	}
+	go func() {
+		for d := range domainsChan {
+			domainsComplete = append(domainsComplete, d)
+			wg.Done()
```

Comment:
you should not play with a non-pointer wg from closure scopes
you should pass pointer to wg as argument to goroutine closure, or declare wg as a pointer to sync.WaitGroup